### PR TITLE
New gatsby based nodeschool template

### PR DESCRIPTION
**Pull request by halkeye**
If merged, I'll move the random todo items to issues
More ideas:
* Move our random markdown files to real pages (so nodeschool.io/vancouver/hosting.html)

Really out there ideas, mostly as examples of what could be done
* travis ci support (Its using my jenkins right now)
* circle ci support (Its using my jenkins right now)
* github actions support (Its using my jenkins right now) (if can enable for the org)

### CODE REVIEW

**Comment by victorperin** [comment]
Thanks guys, this is awesome! Can we create a generator or something like that to other chapters adopt? (i'm thinking to use it in Campinas and São Paulo chapters).

**Comment by halkeye** [comment]
> Thanks guys, this is awesome! Can we create a generator or something like that to other chapters adopt? (i thinking to use it in Campinas and São Paulo chapters).

I would love to do that, I've started down that road by separating chapter specific stuff into data/ and src/ being more the template

**Comment by halkeye** [comment]
Oooooh https://www.gatsbyjs.org/blog/2018-11-11-introducing-gatsby-themes/

**Comment by Qard** [comment]
It's looking really good! Nice work, Gavin. 🎉

**Comment by halkeye** [comment]
So making a theme is actually pretty easy, data can be local, components can actually be overridden

Much easier to upstream fixes to all chapters that join, its versioned so people don't have to take advantage.

HRMMMMM

```
 (ssh) odin   master ●  …/halkeye/test/vancouver  for i in gatsby-config.js package.json; do echo $i; cat $i; done
gatsby-config.js
module.exports = {
  siteMetadata: {
    title: `Gatsby Default Starter`,
    description: `Kick off your next, great Gatsby project with this default starter. This barebones starter ships with the main Gatsby configuration files you might need.`,
    author: `@gatsbyjs`,
  },
  plugins: [
    {
      resolve: "gatsby-theme-nodeschool",
      options: {},
    },
  ],
}
package.json
{
  "name": "gatsby-starter-default",
  "private": true,
  "description": "A simple starter to get up and developing quickly with Gatsby",
  "version": "0.1.0",
  "author": "Kyle Mathews <mathews.kyle@gmail.com>",
  "dependencies": {
    "gatsby": "^2.17.6",
    "gatsby-source-filesystem": "^2.1.35",
    "gatsby-theme-nodeschool": "github:halkeye/nodeschool-vancouver-gatsby"
  },
  "license": "MIT",
  "scripts": {
    "build": "gatsby build",
    "develop": "gatsby develop",
    "format": "prettier --write \"**/*.{js,jsx,json,md}\"",
    "start": "npm run develop",
    "serve": "gatsby serve",
    "test": "echo \"Write tests! -> https://gatsby.dev/unit-testing\" && exit 1"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/gatsbyjs/gatsby-starter-default"
  },
  "bugs": {
    "url": "https://github.com/gatsbyjs/gatsby/issues"
  }
}
```

**Comment by victorperin** [comment]
That's pretty simple, I loved!
If you create a repo, please, link it here, so I can start helping. :)

**Comment by halkeye** [comment]
So making some progress converting it to a theme
https://github.com/halkeye/gatsby-theme-nodeschool/tree/master/example

it needs some work, but is kinda working. I'll probably update the PR in a couple days

**Comment by victorperin** [comment]
Awesome. I need to learn more about those gatsby themes.
Is it possible to customize other contents? Like, a lot of other chapters are not in english.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,96 @@
...
+twitter: <optional twitter id>
+```
+
+We usually like mentors to have come to a session before, and read up on the https://github.com/nodeschool/organizers/wiki/Event-Mentor-Best-Practices#on-being-a-mentor section on the nodeschool wiki.
+
+# Template
+
+This is heavily based off the nodeschool toronto template. It used jekyll which when I tried it wouldn't compile properly, and as a node group, figured all node was better.
+
+## Install all depenendancies
```
Comment (keywordnew) [inline README.md]:
```suggestion
## Install all dependencies
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,96 @@
...
+name: Name
+github: <github id>
+twitter: <optional twitter id>
+```
+
+We usually like mentors to have come to a session before, and read up on the https://github.com/nodeschool/organizers/wiki/Event-Mentor-Best-Practices#on-being-a-mentor section on the nodeschool wiki.
+
+# Template
+
+This is heavily based off the nodeschool toronto template. It used jekyll which when I tried it wouldn't compile properly, and as a node group, figured all node was better.
```
Comment (keywordnew) [inline README.md]:
```suggestion
This is heavily based off the NodeSchool Toronto template. It used Jekyll which when I tried it wouldn't compile properly, and as a Node.js group, figured all Node.js was better.
```

File: data/mentors/manil.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+name: Manil
+twitter: keywordnew
+github: chowdhurian
```
Comment (keywordnew) [inline data/mentors/manil.yaml]:
```suggestion
github: keywordnew
```

**Review by keywordnew — APPROVED** [review]
This is looking great! 🎉

Left a couple of suggestions but already LGTM.

File: README.md (hunk 2 above)
Comment (halkeye) [reply README.md]: oh man I tried to remember to capitalize all the nodeschools, but didn't get all of them

**Comment by halkeye** [comment]
@keywordnew okay i've committed all your suggestions. I'm going to merge this PR as is, and then moving forward switch to the new gatsby theme I created (also merged your suggestions there).

Builds will be done by my jenkins server until github actions goes GA on the 15(?)th

**Pull request merged**